chapter_6/exercise_6_03/cross_referencer.c

```
@@ -1,250 +1,184 @@
-      high = mid - 1;
-    }
-    else if (cond > 0)
-    {
-      low = mid + 1;
+struct tree_node *add_to_tree(struct tree_node *node_p, char *word,
+                              size_t line_number) {
+    int cond;
+
+    if (node_p == NULL) {
+        node_p = (struct tree_node *)malloc(sizeof(struct tree_node));
+        node_p->line_numbers = add_to_list(node_p->line_numbers, line_number);
```

Cppcheck (2.18.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)